lib/src/metta/runner/stdlib_math.rs

```diff
@@ -0,0 +1,730 @@
+    }
+
+    fn as_execute(&self) -> Option<&dyn CustomExecute> {
+        Some(self)
+    }
+}
+
+impl CustomExecute for PowiMathOp {
+    fn execute(&self, args: &[Atom]) -> Result<Vec<Atom>, ExecError> {
+        let arg_error = || ExecError::from("powi-math expects two arguments: number (base) and number (power)");
+        let base = Into::<f64>::into(args.get(0).and_then(Number::from_atom).ok_or_else(arg_error)?);
+        let pow = Into::<i64>::into(args.get(1).and_then(Number::from_atom).ok_or_else(arg_error)?);
+        let res = base.powi(pow.try_into().unwrap());
+        Ok(vec![Atom::gnd(Number::Float(res))])
+    }
+}
+
+#[derive(Clone, Debug)]
+pub struct PowfMathOp {}
+
+grounded_op!(PowfMathOp, "powf-math");
+
+impl Grounded for PowfMathOp {
+    fn type_(&self) -> Atom {
+        Atom::expr([ARROW_SYMBOL, ATOM_TYPE_NUMBER, ATOM_TYPE_NUMBER, ATOM_TYPE_NUMBER])
+    }
+
+    fn as_execute(&self) -> Option<&dyn CustomExecute> {
+        Some(self)
+    }
+}
+
+impl CustomExecute for PowfMathOp {
+    fn execute(&self, args: &[Atom]) -> Result<Vec<Atom>, ExecError> {
+        let arg_error = || ExecError::from("powf-math expects two arguments: number (base) and number (power)");
+        let base = Into::<f64>::into(args.get(0).and_then(Number::from_atom).ok_or_else(arg_error)?);
+        let pow = Into::<f64>::into(args.get(1).and_then(Number::from_atom).ok_or_else(arg_error)?);
+        let res = base.powf(pow.try_into().unwrap());
+        Ok(vec![Atom::gnd(Number::Float(res))])
+    }
+}
+
+#[derive(Clone, Debug)]
+pub struct SqrtMathOp {}
+
+grounded_op!(SqrtMathOp, "sqrt-math");
+
+impl Grounded for SqrtMathOp {
+    fn type_(&self) -> Atom {
+        Atom::expr([ARROW_SYMBOL, ATOM_TYPE_NUMBER, ATOM_TYPE_NUMBER])
+    }
+
+    fn as_execute(&self) -> Option<&dyn CustomExecute> {
+        Some(self)
+    }
+}
+
+impl CustomExecute for SqrtMathOp {
+    fn execute(&self, args: &[Atom]) -> Result<Vec<Atom>, ExecError> {
+        let arg_error = || ExecError::from("sqrt-math expects one argument: number");
+        let input = Into::<f64>::into(args.get(0).and_then(Number::from_atom).ok_or_else(arg_error)?);
+        Ok(vec![Atom::gnd(Number::Float(input.sqrt()))])
+    }
+}
+
+#[derive(Clone, Debug)]
+pub struct AbsMathOp {}
+
+grounded_op!(AbsMathOp, "abs-math");
+
+impl Grounded for AbsMathOp {
+    fn type_(&self) -> Atom {
+        Atom::expr([ARROW_SYMBOL, ATOM_TYPE_NUMBER, ATOM_TYPE_NUMBER])
+    }
+
+    fn as_execute(&self) -> Option<&dyn CustomExecute> {
+        Some(self)
+    }
+}
+
+impl CustomExecute for AbsMathOp {
+    fn execute(&self, args: &[Atom]) -> Result<Vec<Atom>, ExecError> {
+        let arg_error = || ExecError::from("abs-math expects one argument: number");
+        let input = Into::<f64>::into(args.get(0).and_then(Number::from_atom).ok_or_else(arg_error)?);
+        Ok(vec![Atom::gnd(Number::Float(input.abs()))])
```

Comment:
`abs` in Rust is defined for both `i64` and `f64` thus you can again check `Number` variant to select proper implementation.

Reply:
Difference between abs for int and abs for float is in performance?

Reply:
No, just because I think it is better to keep `i64` as `i64` not promote it to `f64`. Promoting it may lead to the precision loss.

Reply:
Ok.